ai_spider/stats.py

```diff
@@ -78,49 +157,29 @@ def load(self, dct):
 # 1 == not skewed (prefer first, but the rest are even)
 
 POWER = 2
-
-
-class StatsStore:
-    def __init__(self):
-        self.update_queue = UniqueQueue(key=lambda el: el[0])
-        self.update_thread = Thread(daemon=True, target=self.update_loop)
-        self.update_thread.start()
-
-    def stop(self, join=True):
-        self.update_queue.put([None, None])
-        if join:
-            self.update_thread.join()
-
-    def update_loop(self):
-        while True:
-            try:
-                key, val = self.update_queue.get(timeout=4)
-                if key is None:
-                    break
-                self._update(key, val)
-            except Empty:
-                pass
-
-    def update(self, key: str, val: dict):
-        self.update_queue.put([key, val])
-
-    def _update(self, key, val):
-        raise NotImplementedError
-
-    def get(self, key: str) -> dict:
-        raise NotImplementedError
-
-
-class MySqlStat
+    
 class StatsContainer:
     ALL_KEY = "<all>"
 
     def __init__(self, alpha=STATS_EMA_ALPHA, key=None, store: StatsStore = None):
-        self.worker_stats: dict[str, StatsWorker] = defaultdict(lambda: StatsWorker(alpha))
-        self.all = StatsWorker(alpha)
+        self.store = store
```

Comment:
again, the container should not care about the structure of an abstract store.

Reply:
I see you have this line as well on your branch: https://github.com/ArcadeLabsInc/queenbee/blob/4ff3024415a6059ea11f71b157c4b2ff76830c4c/ai_spider/stats.py#L123

I am a bit confused if you're suggestion removing this line entirely or suggestion the store should be used in a different way

Reply:
using the store is fine.   but introspecting on it is not.